### PR TITLE
complete task 7 - /api/articles/:article_id

### PR DESCRIPTION
**Pull request by jothwas**
Added the following for task #7:

- Updated models, controllers and app to satisfy the patch request.
- Created a new file in 'errors' to house all of the custom rejected promises, just incase they become excessive.
- 200:  that an object with the correct properties is being returned for a valid id
- 200: an empty object patch will return the article in full, unamended
- 200: works for positive integers
- 200: works for negative integers
- 400: if invalid id passed
- 404: if id not found in the database

(was questioning whether a test was needed for if the negative integer caused the votes total to drop below zero, but this wasn't explicit / mentioned so I didn't ant to add any functionality not required)

### CODE REVIEW

File: app/app.test.js [hunk 1]
```diff
@@ -26,42 +26,132 @@ describe("app", () => {
...
-        });
+  describe("/api/articles/:article_id", () => {
+    describe("GET", () => {
+      test("status: 200 - response with an object of the following fields for a specific article - author (taken from the username in users table), title, article_id, body, topic, created_at, votes", () => {
+        return request(app)
+          .get("/api/articles/1")
+          .expect(200)
+          .then(({ body }) => {
+            expect(body).toEqual(
+              expect.objectContaining({
```
Comment (daviidbartlett) [inline app/app.test.js:37]: can we respond consistently with an object like {article:{...}}

File: app/app.test.js [hunk 2]
```diff
@@ -26,42 +26,132 @@ describe("app", () => {
...
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Bad request - invalid input");
+          });
+      });
+      test("status: 404 - responds with error message if user attempts to search for an id that does not exist", () => {
+        return request(app)
+          .get("/api/articles/938475")
+          .expect(404)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Not found, please check the url and try again.");
```
Comment (daviidbartlett) [inline app/app.test.js]: maybe just say that the article isn't found?

File: app/app.test.js [hunk 3]
```diff
@@ -26,42 +26,132 @@ describe("app", () => {
...
-        });
+    describe("PATCH", () => {
+      test("status: 200 - returns an object containing the specific fields requested", () => {
+        return request(app)
+          .patch("/api/articles/1")
+          .send({ inc_votes: 1 })
+          .expect(200)
+          .then(({ body }) => {
+            expect(body).toEqual(
+              expect.objectContaining({
```
Comment (daviidbartlett) [inline app/app.test.js:75]: again, wrap in object please

File: app/app.test.js [hunk 4]
```diff
@@ -26,42 +26,132 @@ describe("app", () => {
...
+          });
+      });
+      test("status: 404 - responds with error message if user attempts to search for an id that does not exist", () => {
+        return request(app)
+          .patch("/api/articles/938475")
+          .expect(404)
+          .then(({ body: { msg } }) => {
+            expect(msg).toBe("Not found, please check the url and try again.");
+          });
+      });
```
Comment (daviidbartlett) [inline app/app.test.js:164]: missing invalid body? eg. `{inc_votes:"banana"}`

File: errors/rejected-promises.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+exports.rejectedPromise404 = () => {
+  return Promise.reject({
+    status: 404,
+    msg: "Not found, please check the url and try again.",
```
Comment (daviidbartlett) [inline errors/rejected-promises.js]: you may want to allow this to take an arg to change the error message for each rejection 😄

**Review by emilyb93 — APPROVED** [review]

**Pull request merged**